src/win/fs.c

```diff
@@ -70,8 +70,12 @@
     req->sys_errno_ = (sys_errno);                                          \
   } while (0)
 
-#define VERIFY_FD(fd, req)                                                  \
-  if (fd == -1) {                                                           \
+#define VERIFY_HANDLE(hFile, req)                                           \
+  if (hFile == UV_STDIN_FD ||                                               \
+      hFile == UV_STDOUT_FD ||                                              \
+      hFile == UV_STDERR_FD)                                                \
+    hFile = GetStdHandle((DWORD)(uintptr_t) hFile);                         \
```

Comment:
by "double case" do you mean, "do we need to check the return value of GetStdHandle"?

`GetStdHandle` can fail if the application doesn't have a stdio stream (for example, if it was linked with `/SUBSYSTEM:windows`)